### ISSUE TITLE
Handling of a large amount of files and directorys

### ISSUE THREAD

**Issue by dfarin** [meganz/MEGAcmd#260]
Hiho, im testing mega.nz for my agency and try to figure out how good its working as an additional backup space. Till now i cant figure out how i can handle a large amount of files and directorys.
The best option, mega-cmd, isnt working well for now, "put" has no feedback if its still working or just hung up. If i try webdav i get a lot of timeouts while i use rsync and at least "ftp" works for about 4 hours and then mega-cmd-server crashes.
While testing ftp i got some curios bugs, while uploading some files it runs in an loop and with transfers i can see mega-cmd-ftp is uploading the same file again and again and again ...  the next bug is, after an upload was complete i dont get the whole dirlist back. Even after reload and start/stop there were 3 directorys missing. (They are on the mega server, they just dont get listet in megacmd)
After uploading the directorys again with mega-sync (windows) i see them in cmd too.
The last one, mega(cmd/sync/linux/win) is very ineffective i think, i watched my traffic while uploading form several sources and i can see that mega is making a lot of pauses. I have about 40mbit upload and mega uses about 37mbit, but just in realy little chunks. Here u can see 100 seconds while uploading a big file to mega: https://i.imgur.com/MNpQWQ6.png

My question is, what can i do to make an "fast", stable an continous upload?

Please dont misunderstand, the price is great and for general mega looks like an great service, but if i cant get an stable upload and backup working for some million files and directorys i dont wanna "waste" my time. I work for shops and a lot of other websites and need backups of photos, databases, whole websites.
I hope u understand my terrible english. :)

**Comment by mega-bug** [comment]
I just filed a bug concerning a possible race condition.   Are you by any chance running both megasync (desktop applet) *AND* mega-sync (command line applet) on the same machine using the same drive mappings?    This may be causing corruption.

**Comment by dfarin** [comment]
Nope, i use cmd with my linux system and for windows i use the gui sync client.

**Comment by bozonius** [comment]
dfarin:  Please read my query carefully.  It sounds like you have 2 machines or VM's or one of each, or one dual-boot machine.  My question was if you were running BOTH on the SAME running system.

**Comment by dfarin** [comment]
Hey, i try to make it clear, i have 2 physical machines. 1 desktop with windows 1 ubuntu file server.
For windows i use the sync client, for ubuntu i use mega-cmd.
Both works with network paths, but never the same time on the same path. 
I just use all functions manually for now.

**Comment by bozonius** [comment]
OK, so your situation is clearly different than mine.  I am still suspicious, though, that having two clients sync'ing the same path could cause file damage, which I have observed and is repeatable.  I have not heard anything on the status of the bug I filed.   File damage is not acceptable, no matter how it is happening.
